Libraries/LibWeb/HTML/NavigationParams.cpp

```diff
@@ -24,6 +25,74 @@ void NavigationParams::visit_edges(Visitor& visitor)
     visitor.visit(reserved_environment);
 }
 
+// https://infra.spec.whatwg.org/#set-append
+inline void append_to_ordered_set(Vector<String>& set, String item)
+{
+    if (!set.contains_slow(item))
+        set.append(move(item));
+}
+
+// https://html.spec.whatwg.org/multipage/document-lifecycle.html#check-a-navigation-response's-adherence-to-x-frame-options
+bool NavigationParams::check_a_navigation_responses_adherence_to_x_frame_options(Navigable* navigable) const
```

Comment:
This method needs a fixme either here or at the call site to pass the extra 2(?) arguments